ethcore/transaction/src/transaction.rs

```diff
@@ -16,596 +18,682 @@
 
 //! Transaction data structure.
 
-use std::ops::Deref;
-use ethereum_types::{H256, H160, Address, U256};
-use error;
-use ethkey::{self, recover, Signature, Secret, Public, public_to_address};
+use crate::{error, Error};
+use ethereum_types::{Address, H160, H256, U256};
+use ethkey::{self, public_to_address, recover, Public, Secret, Signature};
 use hash::keccak;
-use rlp::{self, RlpStream, Rlp, DecoderError, Encodable};
+use rlp::{self, DecoderError, Rlp, RlpStream};
+use std::ops::Deref;
+
+mod legacy;
+pub mod tx_builders;
+pub use self::legacy::{LegacyTransaction, UnverifiedLegacyTransaction};
+pub use self::tx_builders::TransactionWrapperBuilder;
+mod eip2930;
+pub use self::eip2930::{AccessList, AccessListItem, Eip2930Transaction, UnverifiedEip2930Transaction};
+mod eip1559;
+pub use self::eip1559::{Eip1559Transaction, UnverifiedEip1559Transaction};
 
-type Bytes = Vec<u8>;
 type BlockNumber = u64;
+type Bytes = Vec<u8>;
+
+#[derive(Clone, PartialEq)]
+pub enum TxType {
+    Legacy,
+    Type1 = 1,
+    Type2 = 2,
+    Invalid,
+}
```

Comment:
Can you please add doc comments about what each transaction type is?

Reply:
added in [5dbe940](https://github.com/KomodoPlatform/mm2-parity-ethereum/pull/1/commits/5dbe94081c38472bd733c15a1b4f614e6dcc54dd)